pkg/probe/http/request.go

```diff
@@ -113,7 +113,8 @@ func formatURL(scheme string, host string, port int, path string) *url.URL {
 func v1HeaderToHTTPHeader(headerList []v1.HTTPHeader) http.Header {
 	headers := make(http.Header)
 	for _, header := range headerList {
-		headers[header.Name] = append(headers[header.Name], header.Value)
+		key := http.CanonicalHeaderKey(header.Name)
+		headers[key] = append(headers[key], header.Value)
```

Comment:
I think you can do `headers.Add` to handle this as well

Reply:
Updated